eras/alonzo/impl/testlib/Test/Cardano/Ledger/Alonzo/ImpTest.hs

```diff
@@ -501,8 +504,48 @@ expectPhase2Invalid ::
   ) =>
   Tx era ->
   ImpTestM era ()
-expectPhase2Invalid tx = do
+expectPhase2Invalid = void . submitPhase2InvalidTx
+
+submitPhase2InvalidTx ::
+  ( HasCallStack
+  , AlonzoEraImp era
+  , InjectRuleFailure "LEDGER" AlonzoUtxosPredFailure era
+  ) =>
+  Tx era ->
+  ImpTestM era (Tx era)
+submitPhase2InvalidTx tx = do
```

Comment:
For the sake of consistency in naming, I think it is worth suffering through a little bit of breakage and major version bump for `cardano-ledger-alonzo`
```suggestion
expectPhase2Invalid_ = void . submitPhase2Invalid

submitPhase2Invalid ::
  ( HasCallStack
  , AlonzoEraImp era
  , InjectRuleFailure "LEDGER" AlonzoUtxosPredFailure era
  ) =>
  Tx era ->
  ImpTestM era (Tx era)
submitPhase2Invalid tx = do
```

Reply:
I'm not sure this naming is good.

1. We have a great many other `expectXXX` functions that return `()` so I don't think it's helpful to add the `_` suffix for this one. There are some in `cardano-ledger-core/testlib` (such as `expectRight`) that return some portion of their inputs, but I think these are confusing because an `expect` function normally returns an `Expectation` or something equivalent to it (ie `MonadIO m => m ()`).

1. For the `submit` function, I added the `Tx` suffix so that it was parallel to `submitTx` which is used widely and is very similar.

Reply:
1. I've updated my suggestion to rename `expectPhase2Invalid` -> `submitPhase2Invalid_`. I didn't even notice the difference, to be honest with you. I do agree that `submit` is better than `expect` in this case as well.

2. There is `submitTx` and `submitTx_`, difference between the two is that latter returns unit. Which is exactly the same pattern here, so I'd like to follow the same pattern for this set of functions as well.

Reply:
Done